### PR TITLE
Generate {application,text}/xml headers on gateway by client's choice

### PR DESCRIPTION
**Pull request by arielshaqed**
If no header is specified, none is explicitly generated.  Then the Go http server generates
one by scanning content, it happens to be text/xml.  If either of these headers is
specified, generate that requested content type.  If (only) other headers are generated,
respond "Not Acceptable".

This does not break backwards compatibility with clients that do not generate such a
header (boto, mc).  This should improve compatibility with a client that explicitly asks for
application/xml (the Akka S3 client alpakka only accepts that Content-Type, so it had better
ask for it).  It *will* break a lying client that requests only anything-but-standard-XML.

### CODE REVIEW

**Comment by arielshaqed** [comment]
Tested by capturing traffic (from the minio/mc client) to our gateway and adding various `Accept:` header values.  (The AWS sigv4 security protocol allows replays, that is so very useful for testing, if horribly badly wrong for actual security...)

Should fix #987 , but ideally wait for customer confirmation that it does before pulling :-)

**Comment by arielshaqed** [comment]
@keimoon could you please take a look, too, and see whether this PR would fix your issue?

**Comment by keimoon** [comment]
Unfortunately, alpakka does not send `Accept` header so this fix does not work for us. Currently we are using an work around (proxy the response to replace the response header) to overcome this issue.

**Review by johnnyaug — APPROVED** [review]

**Comment by arielshaqed** [comment]
Thanks!  Also need to set the default to `application/xml` because @keimoon points out the Alpakka client doesn't pass an `Accepts:` header.  Turns out that passing this header just doesn't happen in the world of S3 clients.  Since S3 and Minio both seem happy to return application/xml, go with that.

File: gateway/handler.go
Comment (ozkatz) [inline gateway/handler.go:396]: Won't this return `application/xml` for non-xml responses? (i.e. `GetObject`?)

Comment (arielshaqed) [reply gateway/handler.go:396]: > Won't this return `application/xml` for non-xml responses? (i.e. `GetObject`?)

Ouch, good catch!  I'll need to figure out how to make go-swagger prefer application/xml :-(

**Comment by arielshaqed** [comment]
Dismissed reviews because of Oz' issue with adding content types to files and heads from the backing store.  Will re-request soon(-ish).

**Comment by arielshaqed** [comment]
@johnnyaug @ozkatz PTAL, I cleaned up behaviour of `Accepts:`, `get-object` is back to ignoring everything except the auto-detected content-type.

File: gateway/handler.go
Comment (johnnyaug) [inline gateway/handler.go:378]: Rephrase comment, or put it somewhere else

File: gateway/handler.go
Comment (johnnyaug) [inline gateway/handler.go:383]: Also rephrase to clarify how this comment is related to this code

**Review by johnnyaug — APPROVED** [review]

**Comment by arielshaqed** [comment]
@ozkatz are we good to go with this?

**Review by ozkatz — APPROVED** [review]
looks good!

**Comment by arielshaqed** [comment]
Thanks!

**Pull request merged**
